core/src/test/java/nl/nn/adapterframework/stream/MessageTest.java

```diff
@@ -95,143 +93,156 @@ protected void testAsByteArray(Message adapter) throws IOException {
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		testAsInputSource(adapter);
 		
-		String captured = writer.toString();
+		String captured = new String(outputStream.toByteArray(), "utf-8");
 		assertEquals(testString, captured);
 	}
 	
 	@Test
-	public void testStreamAsByteArrayCaptured() throws Exception {
+	public void testInputStreamAsByteArrayCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		testAsByteArray(adapter);
 		
-		String captured = writer.toString();
+		String captured = new String(outputStream.toByteArray(), "utf-8");
 		assertEquals(testString, captured);
 	}
 
 	@Test
-	public void testStreamAsStringCaptured() throws Exception {
+	public void testInputStreamAsStringCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		testAsString(adapter);
 		
-		String captured = writer.toString();
+		String captured = new String(outputStream.toByteArray(), "utf-8");
 		assertEquals(testString, captured);
 	}
 
 	@Test
-	public void testStreamClosedButCaptured() throws Exception {
+	public void testInputStreamClosedButCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		adapter.asInputStream().close();
+		adapter.close();
 		
-		String captured = writer.toString();
-		assertEquals("(--MoreBytesAvailable--)["+testString+"]", captured);
+		String captured = new String(outputStream.toByteArray(), "utf-8");
+		assertEquals(testString, captured);
 	}
 
 	@Test
-	public void testStreamPreservedAndCaptured() throws Exception {
+	public void testInputStreamPreservedAndCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		adapter.preserve();
 		
-		String captured = writer.toString();
+		String captured = new String(outputStream.toByteArray(), "utf-8");
 		assertEquals(testString, captured);
+		adapter.close();
```

Comment:
Waarom hier?

Reply:
Anders geeft Eclipse een warning op regel 262: Resource leak: 'adapter' is never closed